glusterd2/commands/peers/deletepeer.go

```diff
@@ -43,13 +43,21 @@ func deletePeerHandler(w http.ResponseWriter, r *http.Request) {
 		restutils.SendHTTPError(ctx, w, status, err)
 		return
 	}
+
 	// You cannot remove yourself
 	if id == gdctx.MyUUID.String() {
 		logger.Debug("request denied, received request to delete self from cluster")
 		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "removing self is disallowed.")
 		return
 	}
 
+	//Check if Node is alive
+	if _, alive := store.Store.IsNodeAlive(id); !alive {
```

Comment:
alive is set by peer in regular interval. What happens to this request if alive state is set and then peer goes down and peer remove is called before the TTL of alive key?

Reply:
@aravindavk  then we can remove peer details from the store if a peer successfully left the cluster.